### DISCUSSION TITLE
app reported as damaged or incomplete on macOS Ventura aarch64 (M1)

### DISCUSSION THREAD

**Discussion by StephFo** [eclipse-tycho/tycho#2915] [Q&A] [answered]
Hi,

I'm using maven / tycho (4.0.3 at the moment) from a while without any issue to build my personal eclipse e4 rcp app.
This apps is running fine under win, linux and macOS intel 64bits.

Recently I had the opportunity to run it on mac equipped with M1 chip (aarch64).

I'm relying on eclipse target platform based on 2022-09 modeling version that supports M1 chip. 
I run smoothly Eclipse IDE same version 2022-09 modeling version on this mac M1 chip.

To build against aarch64 I activated in my pom this <arch>aarch64</arch> and the app is built fine.

I don't have Apple Dev ID, then my app is not signed.

I deactivated Gatekeeper to run it the first time on my macOS intel chip  and it is fine, running without any issue.
I did the same on macOS **M1** chip, but when double clicking on the app, macOS Ventura 13.6 reported : App damaged or incomplete ?

The same  build against <arch>x86_64</arch> is running fine without any report from macOS Ventura 13.6 **intel** chip.

Eclipse IDE runs fine, then I replaced in Eclipse.app the Eclipse folder content( that contains features, plugins and other files) with the one from Myapp.app  --> double clicking on Eclipse icon and the Myapp runs fine. But it is named as Eclipse instead of Myapp.

Then I conclude it is not linked to the java part itself but more about macOS bundle package.

Does anyone have face similar situation ?
Does anyone from Tycho dev team experience same thing on mac M1 ?

Regards,
Stéphane

**Comment by StephFo** [comment, marked as answer]
Writing the issue help to better ask google and I found the solution: no issue at all with the app and tycho way to create macOS bundle package whatever it is aarch64 or x86_64.


The issue is related to macOS security: even if I deactivated Gatekeeper, it is not the right way.

I found the solution there: https://www.reddit.com/r/macsysadmin/comments/13vu7f3/app_is_damaged_and_cant_be_opened_error_on_ventura/?rdt=44382#

running this 2 commands:
xattr -l /path/to/application.app
xattr -dr com.apple.quarantine /path/to/application.app

Magic happens, my (not signed with Apple Dev ID) is now running fine.


By the way, i would be interested in a pointer that highlights how to sign with an Apple Dev ID an app built with Tycho please ?

**Reply by laeubi** [reply]
> By the way, i would be interested in a pointer that highlights how to sign with an Apple Dev ID an app built with Tycho please ?

I think there is nothing out of the box, but you can use e.g. `maven-exec-plugin` to perform certain actions before packaging the product.

**Reply by jcompagner** [reply]
We also ship our product for both aarchs..

but we do notarize and sign the dmg fully and then for us it works fine

**Comment by StephFo** [comment]
> We also ship our product for both aarchs..
> 
> but we do notarize and sign the dmg fully and then for us it works fine

Can you please advise how you create dmg and notarize and sign ? Do you use codesign executable called from maven-exec-plugin ?

Thanks

**Reply by jcompagner** [reply]
No we just generate the tar.gz file with Tycho that is then uploaded to a Mac machine where we run scripts to notarize it and make a signed eng.

So that is all done on a special Mac
